src/parsing.ts

```diff
@@ -87,31 +94,18 @@ export const learnFileId = async (_graph: Graph, filePath: string) => {
 
 export const parseDirectory = async (
   graph: Graph,
-  directory: string,
   fileCallback: (graph: Graph, path: string) => Promise<void>
 ) => {
-  const files = await vscode.workspace.fs.readDirectory(
-    vscode.Uri.file(directory)
+  const files = await vscode.workspace.findFiles(
+    `**/*{${(getFileTypesSetting() as string[]).map((f) => `.${f}`).join(",")}}`
```

Comment:
This could be improved if `getFileTypesSetting()` could return ".md" instead of "md"!